### PR TITLE
Cal/refactor bounding volumes

### PR DESCRIPTION
**Pull request by McCallisterRomer**
-Added Sphere collider
-Added ColliderType & BoundingVolume to deal with different collider types
-Moved default assets from /project to new directory /nc/resources

### CODE REVIEW

File: nc/source/component/Collider.cpp [hunk 1]
```diff
@@ -59,14 +59,21 @@ namespace nc
         if(!std::exchange(m_selectedInEditor, false))
             return;
 
-        m_widgetModel.SetTransformationMatrix(m_colliderMatrix);
+        /** @todo right multiply by offset */
+        m_widgetModel.SetTransformationMatrix
+        (
+            DirectX::XMMatrixScaling(m_scale.x, m_scale.y, m_scale.z) *
```
Comment (jaremieromer) [inline nc/source/component/Collider.cpp]: I prefer single line here

File: nc/source/ecs/EcsApiImpl.cpp [hunk 2]
```diff
@@ -16,7 +16,7 @@ namespace nc
 
     EntityHandle CreateEntity(EntityInfo info)
     {
-        IF_THROW(info.scale == Vector3::Zero(), "CreateEntity - scale cannot be Vector3::Zero");
+        IF_THROW(info.scale.x == 0.0f || info.scale.y == 0.0f || info.scale.z == 0.0f, "CreateEntity - Invalid scale(elements cannot be 0)");
```
Comment (jaremieromer) [inline nc/source/ecs/EcsApiImpl.cpp]: I still want to find a way to extract this elsewhere

File: nc/source/physics/CollisionSystem.cpp [hunk 3]
```diff
@@ -21,16 +21,18 @@ namespace nc::physics
...
-        DirectX::BoundingOrientedBox unit, a, b;
+        Collider::BoundingVolume firstVolume, secondVolume;
         const auto count = colliders.size();
-        for(size_t i = 0u; i < count; ++i)
+        for(size_t first = 0u; first < count; ++first)
         {
-            unit.Transform(a, colliders[i]->GetTransformationMatrix());
-            for(size_t j = i + 1; j < count; ++j)
+            firstVolume = colliders[first]->GetBoundingVolume();
+            for(size_t second = first + 1; second < count; ++second)
```
Comment (jaremieromer) [inline nc/source/physics/CollisionSystem.cpp:29]: easier to follow now with the rename

File: nc/source/physics/CollisionSystem.cpp [hunk 4]
```diff
@@ -92,4 +94,4 @@ namespace nc::physics
     {
         m_previousCollisions.clear();
     }
-}
\ No newline at end of file
+} // namespace nc::phsyics
```
Comment (jaremieromer) [inline nc/source/physics/CollisionSystem.cpp:97]: Are we doing this now?

**Review by jaremieromer — APPROVED** [review]

File: nc/source/ecs/EcsApiImpl.cpp (hunk 2 above)
Comment (McCallisterRomer) [reply nc/source/ecs/EcsApiImpl.cpp]: Done, there is a function to do the validation, and it is called only in Transform/Collider.

**Pull request merged**